Marlin/src/feature/runout.h

```diff
@@ -332,11 +335,11 @@ class FilamentSensorBase {
 
   class RunoutResponseDebounced {
     private:
-      static constexpr int8_t runout_threshold = 5;
+      static constexpr int8_t runout_threshold = FILAMENT_RUNOUT_THRESHOLD;
       static int8_t runout_count;
     public:
       static inline void reset()                                  { runout_count = runout_threshold; }
-      static inline void run()                                    { runout_count--; }
+      static inline void run()                                    { if (runout_count >= 0) runout_count--; }
```

Comment:
Please demonstrate a case where `runout_count` can fall below `-127` and this extra overhead will be justified. Remember: we run Marlin on slow AVR processors, and we have to justify every added cycle. If this is not 100% proved to be needed, then it is not wanted.

Reply:
- so first : this code is only executed when there is no more filament in the extruder.
- second : this "adds" one arm instruction, the arm asm has a special instruction to avoid conctional jumps.. sp second one (dec) will be skipped most of the time
- third : I had to code a getter and display the value on the ui to notice that.... I even had to increase it to big int to see the countdown.... and you hardly can see the counter decrease, which is too fast

Reply:
![image](https://user-images.githubusercontent.com/145119/57208735-9d57b980-6fd5-11e9-8f66-56fb9b47cf41.png)

Reply:
Makes sense!